src/accordion.ts

```diff
@@ -0,0 +1,185 @@
+import { LitElement, html } from "lit";
+import {
+    customElement,
+    property,
+    queryAssignedElements,
+} from "lit/decorators.js";
+
+@customElement("aalam-accordion")
+export class AalamAccordion extends LitElement {
+    
+    @property()
+    togglesel: string = ".acc-title";
+
+    @property()
+    animationDur: number = 200;
+
+    @property()
+    bodysel: string = ".acc-body";
+
+    @property({ type: Boolean })
+    nocloseall: boolean = false;
+
+    @property({ type: Boolean })
+    nomultiple: boolean = false;
+
+    @property({ type: String })
+    activecls: string = "acc-active";
+
+    private openItems: Element[] = [];
+
+    @queryAssignedElements({ flatten: true })
+    private items!: Element[];
+
+    private processedItems = new Set<HTMLElement>();
+
+    override render() {
+        return html`<slot @slotchange=${this.handleSlotChange} @transitionend=${this._handleTransitionEnd} ></slot>`;
+    }
+    private _handleTransitionEnd(event: any) {
+        const body = event.target;
+        if (body.style.height == "0px") {
+            body.style.display = "none";
+        }
+        body.style.height = "";
+        body.style.overflow = "";
+        body.style.transitionProperty = "";
+        body.style.transitionDuration = "";
+        body.style.transitionTimingFunction = "";
+    }
+
+    override attributeChangedCallback( name: string, _old: string | null, value: string | null ): void {
+        super.attributeChangedCallback(name, _old, value);
+        if (name === "nomultiple") {
+            if (this.nomultiple && this.openItems.length > 1) {
+                for (let i = 1; i <= this.openItems.length; i++) {
+                    this._closeItem(this.openItems[i - 1], i);
+                }
+            }
+        } else if (name === "nocloseall") {
+            if (
+                this.nocloseall &&
+                this.openItems.length === 0 &&
+                this.items.length > 0
+            ) {
+                this._openItem(this.items[0], 0);
+            }
+        }
+    }
+
+    private handleSlotChange(e: Event) {
+        const slot = e.target as HTMLSlotElement;
+        const newItems = slot.assignedElements({
+            flatten: true,
+        }) as HTMLElement[];
+
+        newItems.forEach((item, index) => {
+            if (this.processedItems.has(item)) {
+                return;
+            }
+
+            this.processedItems.add(item);
```

Comment:
I believe we might have to check for any removed elements as well. Say if there is an item removed, we would still hold the reference in the processed item and will not free up.